test/workflow_test.go

```diff
@@ -170,7 +170,7 @@ func (w *Workflows) ActivityRetryOnHBTimeout(ctx workflow.Context) ([]string, er
 	}
 
 	elapsed := workflow.Now(ctx).Sub(startTime)
-	if elapsed < 5*time.Second {
+	if elapsed < opts.ScheduleToCloseTimeout-opts.RetryPolicy.MaximumInterval {
```

Comment:
I would add description of this PR (may be with some mods) as comment here.

Reply:
done